pkg/offline/error.go

```diff
@@ -0,0 +1,20 @@
+package offline
+
+import (
+	"fmt"
+)
+
+// ErrOpenDB is an error returned when the database cannot be opened.
+type ErrOpenDB struct {
+	Err error
+}
+
+// Error method to implement error interface.
+func (e ErrOpenDB) Error() string {
+	return e.Err.Error()
+}
+
+// Message method to implement wakaerror.Error interface.
+func (e ErrOpenDB) Message() string {
+	return fmt.Sprintf("failed to open db connection: %s", e.Err)
```

Comment:
let's remove the word `connection` here because people will think it means over a network when reading logs.